backend/src/src-wsimport/src/main/java/org/eclipse/sw360/wsimport/thrift/ThriftUploader.java

```diff
@@ -236,15 +221,31 @@ private ReleaseRelation createReleaseRelation(WsLibrary wsLibrary, User sw360Use
     private Set<ReleaseRelation> createReleases(WsProject wsProject, User sw360User, TokenCredentials tokenCredentials) {
         WsLibrary[] libraries = null;
         try {
-            libraries =  new WsImportProjectService().getProjectLicenses(wsProject.getProjectToken(), tokenCredentials);
+            libraries =  new WsImportService().getProjectLicenses(wsProject.getProjectToken(), tokenCredentials);
         } catch (JsonSyntaxException jse) {
             LOGGER.error(jse);
         }
-
-        List<WsLibrary> libraryList = Arrays.asList(libraries);
-
-        if (libraryList == null) {
+        List<WsLibrary> libraryList;
+        if (libraries == null) {
             return ImmutableSet.of();
+        } else {
+            libraryList = new ArrayList<>(Arrays.asList(libraries));
+        }
+
+        WsProjectVitalInformation[] projectVitalInformations = null;
+        List<String> projectNameList = new ArrayList<>();
+        try {
+            projectVitalInformations =  new WsImportService().getOrganizationalProjectVitals(tokenCredentials);
+        } catch (JsonSyntaxException jse) {
+            LOGGER.error(jse);
+        }
+        if(projectVitalInformations != null) {
+            projectNameList = Arrays.stream(projectVitalInformations)
+                    .map(WsProjectVitalInformation::getName)
+                    .collect(Collectors.toList());
+        }
+        for (String project : projectNameList) {
+            libraryList.removeIf(x -> x.getName().equals(project.contains(" ") ? project.split(" ")[0] : project));
```

Comment:
This line is not easy to read. the `project` string can contain spaces?
If it contains spaces why you only want to have the first word (index 0)?
Is it possible that `project` contains leading spaces like "   myProject". 
In your case you will only find " " names and remove these from your array.
Maybe you should write two lines here. First create your remove String variable, after this remove it from the array.

In addition, please use "l" like `library` instead of "x" in your `libraryList.removeIf" function.

Reply:
yes, that is true. I will write this out to two lines to improve readability